src/renderers/gl/renderer.cpp

```diff
@@ -377,24 +377,35 @@ auto mrg::Renderer::render(mg::RenderableList const& renderables) const -> std::
 
 void mrg::Renderer::draw(mg::Renderable const& renderable) const
 {
+    auto const texture = gl_interface->as_texture(renderable.buffer());
     auto const clip_area = renderable.clip_area();
     if (clip_area)
     {
         glEnable(GL_SCISSOR_TEST);
-        glScissor(
-            clip_area.value().top_left.x.as_int() -
-                viewport.top_left.x.as_int(),
-            viewport.top_left.y.as_int() +
+
+        int y;
```

Comment:
It would be nice to initialise y directly.
```suggestion
        auto const invert_clip_area = ....;
        auto const scissor_y = invert_clip_area ?
            (viewport.top_left.y + viewport.size.height).as_int() - (clip_area.value().top_left.y - clip_area.value().size.height).as_int() :
            clip_area.value().top_left.y.as_int();
```